src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -94,7 +94,9 @@ void AddSubMenuRequest::Run() {
   const std::string& menu_name =
       received_msg_params[strings::menu_name].asString();
 
-  if (app->IsSubMenuNameAlreadyExist(menu_name)) {
+  const int32_t parent_id = received_msg_params[strings::parent_id].asInt();
```

Comment:
```suggestion
  const uint32_t parent_id = received_msg_params[strings::parent_id].asUInt();
```
Also, you should check if this value exists, otherwise a bad value is likely returned (this would only affect the case where an older proxy is sending the message, otherwise the value would be filled with the default value of 0).